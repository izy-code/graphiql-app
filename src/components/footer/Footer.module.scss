.footer {
  display: flex;
  flex-flow: column nowrap;
  gap: 10px;
  box-sizing: border-box;
  width: 100%;
  margin: 0 auto;
  padding: 10px 0;
  color: var(--color-text-dark);
  border-top: 1px solid var(--color-border);
}

.links {
  display: flex;
  flex-flow: row nowrap;
  justify-content: space-around;
  align-items: center;
  width: 100%;
  margin: 0 auto;
  font-size: 20px;

  &.small {
    width: 65%;
    padding-bottom: 10px;
    border-bottom: 1px solid var(--color-border);
  }
}

.footer_creators {
  margin: 0;
  padding: 0;
}

.footer_image {
  width: 50px;
  height: 50px;
}

.footer_items {
  width: 105px;
  text-align: center;
}
