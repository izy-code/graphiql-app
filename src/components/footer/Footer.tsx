import Image from 'next/image';
import type { ReactNode } from 'react';

import logo from '../../assets/images/rss-logo.png';
import styles from './Footer.module.scss';

export function Footer(): ReactNode {
  return (
    <footer className={styles.footer}>
      <div className={styles.links}>
        <a className={styles.footer_items} href="https://github.com/">
          Our project
        </a>
        <p>2024</p>
        <div className={styles.footer_items}>
          <Image className={styles.footer_image} src={logo} alt="logo" />
        </div>
      </div>
    </footer>
  );
}
