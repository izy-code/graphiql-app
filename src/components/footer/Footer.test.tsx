import { render } from '@testing-library/react';

import { Footer } from '@/components/footer/Footer';

describe('Footer Component', () => {
  it('renders correctly', () => {
    const { container } = render(<Footer />);

    expect(container).toMatchSnapshot();
  });
});
