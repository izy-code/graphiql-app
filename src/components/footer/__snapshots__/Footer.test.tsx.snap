// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > renders correctly 1`] = `
<div>
  <footer
    class="_footer_660fbd"
  >
    <div
      class="_links_660fbd _devs_660fbd"
    >
      <a
        class="_link_660fbd"
        href="https://github.com/izy-code"
      >
        izy-code
      </a>
      <a
        class="_link_660fbd"
        href="https://github.com/VadimKol"
      >
        vadimkol
      </a>
      <a
        class="_link_660fbd"
        href="https://github.com/BodnarAlex"
      >
        bodnaralex
      </a>
    </div>
    <div
      class="_links_660fbd"
    >
      <a
        class="_github_660fbd"
        href="https://github.com/izy-code/graphiql-app"
      />
      <p
        class="_year_660fbd"
      >
        2024
      </p>
      <a
        class="_rss_660fbd"
        href="https://rs.school/courses/"
      />
    </div>
  </footer>
</div>
`;
